lib/motion-flow/loader.rb

```diff
@@ -0,0 +1,53 @@
+$:.unshift("/Library/RubyMotion/lib")
+
+module Motion
+  class Flow
+    def self.load_library(lib_name)
+      lib_dir = File.join(File.dirname(__FILE__), '../../flow', lib_name)
+
+      case Motion::Project::App.template
+      when :android
+        platform = 'android'
+      when :ios, :tvos, :osx, :'ios-extension'
+        platform = 'cocoa'
+      else
+        raise "Project template #{Motion::Project::App.template} not supported by Flow"
+      end
+
+      Motion::Project::App.setup do |app|
+        app.files.concat(Dir.glob(File.join(lib_dir, '*.rb')))
+        app.files.concat(Dir.glob(File.join(lib_dir, platform, '*.rb')))
+
+        if platform == 'android'
+          abis = %w{armeabi-v7a x86}
+          if abis.all? { |x| File.exist?(File.join(lib_dir, 'android', x)) }
+            abis.each do |abi|
+              app.libs[abi] += Dir.glob(File.join(lib_dir, 'android', abi, "*.a")).map { |x| "\"#{x}\""}
+            end
+          end
+
+          # TODO: figure out which of these vendor libs are required for each Flow lib. Figure out how to selectively add these only if they haven't already been added by the main project.
```

Comment:
I would feel more comfortable getting some feedback on this section before I merge this PR.